test/yaml_test_suite_test.gleam

```diff
@@ -0,0 +1,173 @@
+//               let encoded =
+//                 case
+//                   case yaml {
+//                     String(string) -> string
+//                     _ -> panic as "Yaml must be string"
+//                   }
+//                   |> decode
+//                 {
+//                   Ok(ok) ->
+//                     ok
+//                     |> io.debug
+//                   Error(_) -> panic as "Failed decoding yaml"
+//                 }
+//                 |> encode
+
+//               let encoded_without_start =
+//                 case
+//                   case yaml {
+//                     String(string) -> string
+//                     _ -> panic as "Yaml must be string"
+//                   }
+//                   |> decode
+//                 {
+//                   Ok(ok) -> ok
+//                   Error(_) -> panic as "Failed decoding yaml"
+//                 }
+//                 |> cymbal.encode_without_document_start
+
+//               //   case dump {
+//               //     String(string) -> string
+//               //     _ -> panic as "Dump must be string"
+//               //   }
+//               //   |> should.equal(encoded)
+
+//               io.debug(case
+//                 result.unwrap(
+//                   list.key_find(
+//                     case result.unwrap(list.first(array), block([])) {
+//                       Block(block) -> block
+//                       _ -> panic as "Dump must be block"
+//                     },
+//                     "name",
+//                   ),
+//                   string("Name not found"),
+//                 )
+//               {
+//                 String(string) -> string
+//                 _ -> ""
+//               })
+
+//               io.debug(
+//                 "Expected "
+//                 <> case dump {
+//                   String(string) -> string
+//                   _ -> panic as "Dump must be string"
+//                 },
+//               )
+
+//               case
+//                 case dump {
+//                   String(string) -> string
+//                   _ -> panic as "Dump must be string"
+//                 }
+//                 == encoded
+//               {
+//                 True -> io.debug("Got      " <> encoded)
+//                 False -> io.debug("Got      " <> encoded_without_start)
+//               }
+
+//               io.debug(" ")
+
+//               case dump {
+//                 String(string) -> string
+//                 _ -> panic as "Dump must be string"
+//               }
+//               == encoded
+//               || case dump {
+//                 String(string) -> string
+//                 _ -> panic as "Dump must be string"
+//               }
+//               == encoded_without_start
+//             }
+//             Error(_) -> {
+//               False
+//             }
+//           }
+//         Error(_) -> False
+//       }
+//     }
+//     _ -> False
+//   }
+// }
```

Comment:
How come this code is commented out?

Reply:
Their encoded yaml and the encoded yaml returned from cymbal don't match so no tests are passing there atm. It will need some tweaking but it's just commented so I could work on the other tests in the meantime. I will get to it at some point but I haven't been able to because of school/work